src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -233,11 +261,11 @@ IFACEMETHODIMP_(void) FancyZones::WindowCreated(HWND window) noexcept
     if (m_settings->GetSettings().appLastZone_moveWindows)
     {
         auto processPath = get_process_path(window);
-        if (!processPath.empty()) 
+        if (!processPath.empty())
         {
-            INT zoneIndex = -1;
-            LRESULT res = RegistryHelpers::GetAppLastZone(window, processPath.data(), &zoneIndex);
-            if ((res == ERROR_SUCCESS) && (zoneIndex != -1))
+            INT zoneIndex = JSONHelpers::FancyZonesDataInstance().GetAppLastZone(window, processPath.data(), &zoneIndex);

```

Comment:
`zoneIndex` is both used as a return value and reference parameter, that looks like a bug

Reply:
Will be fixed